src/Voron/Impl/Journal/JournalFile.cs

```diff
@@ -149,7 +149,8 @@ public void Write(LowLevelTransaction tx, CompressedPagesResult pages, LazyTrans
 
             if (tx.IsLazyTransaction == false && (lazyTransactionScratch == null || lazyTransactionScratch.HasDataInBuffer() == false))
             {
-                _journalWriter.WritePages(position, pages.Base, pages.NumberOfPages);
+                Debug.Assert(pages.NumberOfPages < int.MaxValue);
```

Comment:
And in general, if the transaction size is bigger than 2GB, it isn't lazy.
Also, you are comparing the number of pages to 2GB, but that isn't the actual size.